### PR TITLE
fix broken column resize 

### PR DESCRIPTION
**Pull request by jpdriver**
- [x] make All Features examples resizable
- [x] use ColumnUtils.getColumn() before performing updates
- [x] add updateMetrics helper to slice in updatedColumn

### CODE REVIEW

File: src/ColumnUtils.js [hunk 1]
```diff
@@ -8,6 +8,15 @@ module.exports = {
     }
   },
 
+  updateMetrics(metrics, idx, column){
```
Comment (malonecj) [inline src/ColumnUtils.js]: we should change the function name to splice as that is what is happening here

File: src/ColumnUtils.js [hunk 2]
```diff
@@ -8,6 +8,15 @@ module.exports = {
...
+    if(Array.isArray(metrics.columns)){
+      metrics.columns.splice(idx, 1, column);
+    }else if (typeof Immutable !== 'undefined') {
```
Comment (malonecj) [inline src/ColumnUtils.js]: this is fine for now - but we should set a flag when the grid is initialised (most probably on the column metrics object) to say whether we are using immutable data or not

**Comment by malonecj** [comment]
needs tests please

**Comment by aqilzeeshan** [comment]
@malonecj please review it.

File: src/__tests__/Header.spec.js [hunk 3]
```diff
@@ -43,11 +43,44 @@ describe('Header Unit Tests', () => {
...
+      testProps.columnMetrics.columns = new Immutable.List(helpers.columns);
+      header = TestUtils.renderIntoDocument(<Header {...testProps}/>);
+    });
+
+    it('should render a default header row', () => {
+      shouldRenderDefaultRow();
   });
 
   it('header row drag start should set resize column state ', () => {
+                  debugger;
```
Comment (malonecj) [inline src/__tests__/Header.spec.js]: remove debugger statement

**Pull request merged**